api/src/main/java/jakarta/data/metamodel/Attribute.java

```diff
@@ -48,8 +48,15 @@ public interface Attribute<T> {
 
     /**
      * Obtain the Java class which declares this entity attribute.
+     * @apiNote This is only guaranteed to be known if a static <code>of</code> method,
+     *          such as {@link BasicAttribute#of(Class, String, Class)}, was used to obtain 
+     *          the instance.
+     *
      * @return the declaring class
+     * @throws UnsupportedOperationException if the declaring type is not known.
      * @since 1.1
      */
-    Class<T> declaringType();
+    default Class<T> declaringType() {
+        throw new UnsupportedOperationException();
```

Comment:
Please, at least put a message on this exception.
I am not a massive fan of this approach, mainly because the interface starts breaking the Liskov Substitution Principle

Reply:
Given that the 1.1 spec can perfectly reasonably *require* that 1.1 providers *do* implement that method, and that there is no expectation that the 1.1 API jar can be used with a 1.0 provider, I don't see much of a problem here.

Is it unfortunate that code generated with 1.0 might fail when you call a method defined by 1.1? I mean, yeah, I guess. But since the solution is just "recompile your program with a 1.1 implementation", I just don't see it as a big deal at all.